tokio-console/Cargo.toml

```
@@ -29,14 +29,27 @@ keywords = [
 [dependencies]
 atty = "0.2"
 console-api = { version = "0.6.0", path = "../console-api", features = ["transport"] }
-clap = { version = "3", features = ["cargo", "derive", "env"] }
+# Use 1.64.0 compatible version of `clap@4.1.14`.
+clap = { version = "=4.1.14", features = [
+    # these are all enabled by default
+    "std", "color", "help", "usage", "error-context", "suggestions",
+    # auto-wrap help text to fit the terminal width
+    "wrap_help",
+    # use Cargo env vars when determining help text
+    "cargo",
+    # enable derive attributes
+    "derive",
+    # parse values from env vars
+    "env"
+] }
```

A couple of things here.

1. Let's keep a single line per dependency the same as the other dependencies and remove the default features as we don't need to specify them
2. From the tags in the clap repo, clap 4.3.24 also has an MSRV of 1.64, so we can use that one. Also, we don't need to fix an exact version, but instead use `~4.3` which means `>= 4.3.0` and `< 4.4.0`.

Mostly this is about trusting the clap maintainers to uphold semver. Default features shouldn't be removed except in a major version change and it's the norm to not update the MSRV in a patch update.

```suggestion
clap = { version = "~4.3", features = ["wrap_help", "cargo", "derive", "env"] }
```

---

Because Cargo doesn't consider the MSRV when resolving the deps. So if we try to use `~4.3`. Then we need to update some other deps to some specific version.

For example:
```console
console on  rustin-patch-clap [!] via 🦀 v1.64.0 
❯ cargo b
    Updating crates.io index
error: package `anstyle-parse v0.2.2` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.64.0

console on  rustin-patch-clap [!] via 🦀 v1.64.0 
❯ cargo update -p anstyle-parse --precise 0.2.1       
    Updating crates.io index
    Updating anstyle-parse v0.2.2 -> v0.2.1

console on  rustin-patch-clap [!] via 🦀 v1.64.0 took 6s 
❯ cargo b                                      
error: package `anstyle v1.0.4` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.64.0

console on  rustin-patch-clap [!] via 🦀 v1.64.0 
❯ cargo update -p anstyle --precise 1.0.1 
    Updating crates.io index
    Updating anstyle v1.0.4 -> v1.0.1

console on  rustin-patch-clap [!] via 🦀 v1.64.0 
❯ cargo b                                
error: package `clap_lex v0.5.1` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.64.0

console on  rustin-patch-clap [!] via 🦀 v1.64.0 

```

---

Removed all comments.

---

That's anoying, even `4.1.14` doesn't work?

---

`4.1.14` works well. Because it doesn't have those deps.

---

OK, then let's use `"^4.1.14"` for the version then.

---

`^4.1.14` same as `4.1.14`: 4.1.14  :=  >=4.1.14, <5.0.0.

So it wouldn't help us to use a compliable version.

---

Sorry! I meant `"~4.1.14"`.

---

Sounds good. Updated!